uaclient/data_types.py

```diff
@@ -11,6 +11,12 @@
 INCORRECT_FIELD_TYPE_ERROR_MESSAGE = (
     'Got value with incorrect type for field "{key}": {nested_msg}'
 )
+INCORRECT_ENUM_VALUE_TYPE_ERROR_MESSAGE = (
+    "Got value {value} of type {value_type} which is not supported in enum"
+)
+INCORRECT_ENUM_VALUE_ERROR_MESSAGE = (
+    "Got value {value} which is not found in enum {enum_class}"
+)
```

Comment:
Let's avoid printing actual values. But we can print the list of all allowed values. Maybe something like:
```
"Value was not found in TestEnum allowed values: [val1, val2, val3]"
```